google/cloud/internal/debug_string_test.cc

```diff
@@ -13,73 +13,79 @@
 // limitations under the License.
 
 #include "google/cloud/internal/debug_string.h"
+#include "absl/strings/string_view.h"
 #include <gmock/gmock.h>
+#include <string>
 
 namespace google {
 namespace cloud {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace internal {
 namespace {
 
+struct SubMessage {
+  double value;
+
+  std::string DebugString(absl::string_view name,
+                          TracingOptions const& options = {},
+                          int indent = 0) const {
+    return DebugFormatter(name, options, indent).Field("value", value).Build();
+  }
+};
+
 TEST(DebugFormatter, SingleLine) {
-  EXPECT_EQ(DebugFormatter(TracingOptions{}.SetOptions("single_line_mode=T"),
-                           "message_name")
+  EXPECT_EQ(DebugFormatter("message_name",
+                           TracingOptions{}.SetOptions("single_line_mode=T"))
                 .Field("field1", 42)
-                .SubMessage("sub_message")
-                .QuotedField("field2", 3.14159)
-                .EndMessage()
+                .SubMessage("sub_message", SubMessage{3.14159})
                 .StringField("field2", "foobar")
                 .Field("field3", true)
                 .Build(),
             R"(message_name {)"
             R"( field1: 42)"
             R"( sub_message {)"
-            R"( field2: "3.14159")"
+            R"( value: 3.14159)"
```

Comment:
I would expect one extra space in the indent. (But I also don't know what protobuf does, and if we are just copying their format)

Reply:
There is always a space after the `{` in single-line mode.